Modules/System/AL Rest Client/src/ALRestClientImpl.Codeunit.al

```diff
@@ -0,0 +1,153 @@
+codeunit 2351 "AL Rest Client Impl."
+{
+    Access = Internal;
+    InherentEntitlements = X;
+    InherentPermissions = X;
+
+    var
+        HttpAuthentication: Interface "Http Authentication";
+        HttpClient: HttpClient;
+        ConnectionError: Boolean;
+        Initialized: Boolean;
+        NotInitializedErr: Label 'The AL Rest Client has not been initialized';
+        EnvironmentBlocksErr: Label 'Environment blocks an outgoing HTTP request to ''%1''.', Comment = '%1 = url, e.g. https://microsoft.com';
+        ConnectionErr: Label 'Connection to the remote service ''%1'' could not be established.', Comment = '%1 = url, e.g. https://microsoft.com';
+        RequestFailedErr: Label 'The request failed: %1 %2', Comment = '%1 = HTTP status code, %2 = Reason phrase';
+        UserAgentLbl: Label 'Dynamics 365 Business Central - |%1| %2/%3', Locked = true, Comment = '%1 = App Publisher; %2 = App Name; %3 = App Version';
+
+    procedure Initialize()
+    begin
+        ClearAll();
+
+        HttpClient.Clear();
+        HttpClient.Timeout := 60000;
+        Initialized := true;
+        SetDefaultUserAgentHeader();
+    end;
+
+    procedure SetBaseAddress(Url: Text)
+    begin
+        AssertInitialized();
+        HttpClient.SetBaseAddress(Url);
+    end;
+
+    procedure GetBaseAddress() Url: Text
+    begin
+        AssertInitialized();
+        Url := HttpClient.GetBaseAddress;
+    end;
+
+    procedure SetTimeOut(TimeOut: Duration)
+    begin
+        AssertInitialized();
+        HttpClient.Timeout := TimeOut;
+    end;
+
+    [NonDebuggable]
+    procedure SetAuthorizationHeader(Value: Text)
+    begin
+        SetDefaultRequestHeader('Authorization', Value);
+    end;
+
+    procedure SetUserAgentHeader(Value: Text)
+    begin
+        SetDefaultRequestHeader('User-Agent', Value);
+    end;
+
+    local procedure SetDefaultUserAgentHeader()
+    begin
+        SetUserAgentHeader(GetUserAgentString());
+    end;
+
+    local procedure GetUserAgentString() UserAgentString: Text
+    var
+        ModuleInfo: ModuleInfo;
+    begin
+        if NavApp.GetCurrentModuleInfo(ModuleInfo) then
```

Comment:
Won't this always be Microsoft + System Application, instead of the caller's module?
The HttpClient already sets a default User-Agent header if I remember correctly, so is this really needed?

Reply:
Yes, this would be the System Application if the caller doesn't overwrite it.
The HttpClient does not set a default User-Agent header, that's why I chose to do it. Just like other modules in the System Application that use the HttpClient.